gems/sorbet-runtime/test/types/opaque.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+module Opus::Types::Test
+  class OpaqueTest < Critic::Unit::UnitTest
+    extend T::Sig
+
+    class OpaqueString < T::Types::Opaque
+      ConcreteType = type_template(fixed: String)
+    end
```

Comment:
```ruby
    class OpaqueString < T::Types::Opaque
      ConcreteType = type_template(fixed: String)
      def foo; end
    end

    sig {params(x: OpaqueString).void}
    def takes_opaque_type(x); x.foo; end
```

^ this would be ok statically but fail at runtime.

Reply:
Ack, good catch

Reply:
Unsure what to do about that. We could enforce at load time that the opaque type doesn't define methods that aren't present on the underlying type, but it feels pretty ugly to have that totally separate from the typechecker. Definitely another big downside to the quick-and-dirty version.